### PR TITLE
LibGUI: Prevent CenterWithinParent Dialogs from appearing offscreen

### PR DESCRIPTION
**Pull request by mikeakers**
When a dialog is created the position is checked against the Desktop's
rect and repositioned to be entirely visible. If the dialog is larger
than the desktop's rect it is just centered.

Before (repro by starting serenity, opening ImageViewer and then File->Open): 
Screen Shot 2022-05-29 at 1 57 16 AM

After:

image

### CODE REVIEW

File: Userland/Libraries/LibGUI/Dialog.cpp [hunk 1]
```diff
@@ -40,6 +41,20 @@ Dialog::ExecResult Dialog::exec()
             auto& parent_window = *static_cast<Window*>(parent());
             if (parent_window.is_visible()) {
                 window_rect.center_within(parent_window.rect());
+                if (window_rect.size().width() < desktop_rect.size().width() && window_rect.size().height() < desktop_rect.size().height()) {
+                    if (window_rect.top() < 20) {
```
Comment (mikeakers) [inline Userland/Libraries/LibGUI/Dialog.cpp:45]: I'm actually not sure why this 20px offset is needed, if there is a constant defined somewhere I can reference for this I would appreciate a pointer

Comment (AtkinsSJ) [reply Userland/Libraries/LibGUI/Dialog.cpp:45]: That might be the height of the window title, I have a feeling that and the other borders aren't included in the window rect. You could test that by using a theme with thicker borders.

Comment (awesomekling) [reply Userland/Libraries/LibGUI/Dialog.cpp:45]: Basically yes. The window frame is not part of its geometry, it's something that the WindowServer renders outside the window rect. :)

**Comment by AtkinsSJ** [comment]
@mikeakers Are you still working on this? Putting a `// FIXME: Use the window title size from the theme instead of 20` in there is fine if you don't want to do that yourself right now.

**Comment by mikeakers** [comment]
@AtkinsSJ Yup, I still intend to work on this. Let me take a run at pulling the metrics I need from the current theme, and if I can't figure that out I'll add a `// FIXME:` and ask for a re-review :)

**Comment by mikeakers** [comment]
Thanks @awesomekling for the pointers during office hours today. Please take another look!

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Very nice! Works well.

One change though: The second commit is a correction to the first one in response to feedback, so please combine them together.

**Comment by mikeakers** [comment]
Squashed commits

**Pull request merged**